amcl/src/amcl_node.cpp

```diff
@@ -794,8 +812,35 @@ void AmclNode::savePoseToServer()
                                   last_published_pose.pose.covariance[6*5+5]);
 }
 
+void AmclNode::deletePoseFromServer()
+{
+  private_nh_.deleteParam("initial_pose_x");
+  private_nh_.deleteParam("initial_pose_y");
+  private_nh_.deleteParam("initial_pose_a");
+  private_nh_.deleteParam("initial_cov_xx");
+  private_nh_.deleteParam("initial_cov_yy");
+  private_nh_.deleteParam("initial_cov_aa");
+  ROS_INFO("Delete pose from server");
+}
+
+bool AmclNode::checkPoseParamOnServer()
+{
+  if(private_nh_.hasParam("initial_pose_x") &&
+      private_nh_.hasParam("initial_pose_y") &&
+      private_nh_.hasParam("initial_pose_a") &&
+      private_nh_.hasParam("initial_cov_xx") &&
+      private_nh_.hasParam("initial_cov_yy") &&
+      private_nh_.hasParam("initial_cov_aa"))
+  {
+    ROS_INFO("Using pose from server since we have all params.");
+    return true;
+  }
+  return false;
+}
+
 void AmclNode::updatePoseFromServer()
 {
+  // default values
```

Comment:
Can we reuse checkPoseParam on sever in this function and reduce duplicate checks.

Reply:
so I did this to maintain code re-usability. There is a way especially in tests where you can just mention few pose params like initial_pose_x and intial_pose_y and for rest of params, use default. That is what `updatePoseFromServer()` does. But we set our own covariance for apriltags and we want ALL params to be set by us. That is why I added this `checkPoseParam` for getting information back to core. Does that justify two functions?